dsc/workflows/base/__init__.py

```diff
@@ -26,37 +26,44 @@ def __init__(
         email_recipients: list[str],
         metadata_mapping: dict,
         s3_bucket: str,
-        s3_prefix: str,
+        batch_id: str,
         collection_handle: str,
         output_queue: str,
     ) -> None:
         """Initialize base instance.
 
         Args:
-            workflow_name: The name of the workflow.
-            submission_system: The system to which item submissions will be sent (e.g.
-            DSpace@MIT)
-            email_recipients: The email addresses to notify after runs of the workflow.
-            metadata_mapping: A mapping file for generating DSpace metadata from the
-            workflow's source metadata.
-            s3_bucket: The S3 bucket containing bitstream and metadata files for the
-            workflow.
-            s3_prefix: The S3 prefix used for objects in this workflow. This prefix does
-            NOT include the bucket name.
-            collection_handle: The handle of the DSpace collection to which submissions
-            will be uploaded
-            output_queue: The SQS output queue used for retrieving result messages from
-            the workflow's submissions.
+            workflow_name (str): The name of the workflow.
+            submission_system (str): The system to which item submissions will be sent
+                (e.g. DSpace@MIT).
```

Comment:
A global comment, not sure that the type hints are needed in the docstrings since they appear in the signature

Reply:
I think I agree.   And, I think it's okay to _informally_ mention the type in the explanation of the argument if it helps reason about how to use.

Reply:
Ah, I see! Whenever I revise a docstring, I tend to let the [autoDocstring extension for VSCode](https://marketplace.visualstudio.com/items?itemName=njpwerner.autodocstring) automatically generate the docstring and I have it set to the [Google Docstring format](https://github.com/NilsJPWerner/autoDocstring/blob/HEAD/docs/google.md), which automatically includes parameter types.

Hmm...do I remove all the typehints from all classes and methods (that I most likely introduced 😅 ) now or ...? I did look at the autoDocstring docs and it seems like I can define a custom template, but I'd like to revisit that process later.

Would love some additional input on how to proceed!

Reply:
Modifying the template sounds good for the future and I would remove them. I think find/replace on something like `(str):` would get a lot of them

Reply:
I'm not philosophically opposed to the docstrings having the types, but they would be my first bet to fall out of sync with the code.  I recall in ABDiff when we did a lot of signature rewriting, that the lengthy docstrings were stale pretty quick.  That's one thing I really like about actual type hints in code, is they _must_ be up-to-date and correct (well, mostly).

Reply:
I removed the docstrings but kept the indentation for multiline parameter definitions since I think it helps with readability!